src/Model/PageTypeGo.cs

```diff
@@ -133,5 +133,18 @@ public override int GetHashCode()
         {
             return Name.GetHashCode();
         }
+
+        /// <summary>
+        /// Updates the page with properties from the passed page. Throws an exception of pages are not equal.
+        /// </summary>
+        /// <param name="page"></param>
+        public void MergeWithPage(PageTypeGo page)
+        {
+            if (!this.Equals(page))
+            {
+                throw new InvalidOperationException("CreatePageableTypeForMethod requires method to be a pageable operation");

```

Comment:
I think the `CreatePageableTypeForMethod` should be `MergeWithPage` yes?

Reply:
ah yea will just use `nameof()`